boards/nordic/nrf52840dk/src/io.rs

```diff
@@ -9,11 +9,26 @@ use nrf52840::gpio::Pin;
 
 use crate::PROCESSES;
 
-struct Writer {
-    initialized: bool,
+enum Writer {
+    WriterUart(/* initialized */ bool),
+    WriterRtt(&'static mut capsules::segger_rtt::SeggerRttMemory<'static>),
 }
 
-static mut WRITER: Writer = Writer { initialized: false };
+static mut WRITER: Writer = Writer::WriterUart(false);
+
+fn wait() {
+    let mut x = 0;
+    for i in 0..5000 {
+        unsafe { core::ptr::write_volatile(&mut x as *mut _, i) };
```

Comment:
```suggestion
        cortexm4::support::nop();
```

If you were relying on the timing here, `nop` is one cycle on Cortex-M while `str` without other oddities is 2, so you may need to up the loop count as well, but I think the explicit `nop` is clearer than reinventing it

Reply:
Thanks for pointing it out! I'll double check that the loop count is correct.

Reply:
I checked the loop again in practice on some panic examples. I could reduce the number of iterations quite a bit - 50 wasn't enough in all cases but 100 seems reasonably stable.